### PR TITLE
Build and push nuget artifacts

### PR DESCRIPTION
**Pull request by pavlovic-ivan**
#### What type of PR is this?
This is an improvement PR
#### What this PR does / why we need it:
This PR add workflow jobs that pack and push nuget artifacts, Armada clients
#### Special notes for your reviewer:
Magefile target that was doing both pack and push (named `PushNuget`), has been refactored to only pack nuget, thus implementing single responsibility principle.

Prerequisites:
- [x] create a protected environment called `nuget-release` with the `NUGET_API_KEY` secret

### CODE REVIEW

**Review by dave-gantenbein — APPROVED** [review]
LGTM

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -99,3 +99,26 @@ jobs:
...
+  push-nuget:
+    name: Push nuget clients
+    needs: validate
+    runs-on: ubuntu-22.04
+    environment: nuget-release
+    steps:
+      - name: Setup the latest .NET 7 SDK
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: 7.0.x
```
Comment (jgiannuzzi) [inline .github/workflows/release.yml:112]: I just realised this step is missing from the `pack-nuget` job! Could you please add it?

Comment (pavlovic-ivan) [reply .github/workflows/release.yml:112]: sure

**Comment by jgiannuzzi** [comment]
The `nuget-release` environment with the `NUGET_API_KEY` secret needs to be created before this PR gets merged. I'm on it!

**Comment by pavlovic-ivan** [comment]
> The `nuget-release` environment with the `NUGET_API_KEY` secret needs to be created before this PR gets merged. I'm on it!

I will add this to the PR desc

**Comment by jgiannuzzi** [comment]
Environment and secret created + PR description updated

File: .github/workflows/release.yml (hunk 1 above)
Comment (pavlovic-ivan) [reply .github/workflows/release.yml:112]: @jgiannuzzi step added

**Review by dave-gantenbein — APPROVED** [review]

**Review by severinson — APPROVED** [review]

**Pull request merged**